iopipe/contrib/eventinfo/event_types.py

```diff
@@ -0,0 +1,154 @@
+from .util import get_nested
+
+
+class EventType(object):
+    def __init__(self, event):
+        self.event = event
+
+    def has_required_keys(self):
+        return all(get_nested(self.event, *key.split('.')) is not None for key in self.required_keys)
+
+    def collect(self):
+        event_info = {}
+        for key in self.keys:
+            value = get_nested(self.event, *key.split('.'))
+            if value is not None:
+                event_info['event-info.%s.%s' % (self.type, key)] = value
+        event_info['event-info.eventType'] = self.type
+        return event_info
+
+
+class ApiGateway(EventType):
+    type = 'apiGateway'
+    keys = [
+        'httpMethod',
+        'path',
+        'requestContext.accountId',
+        'requestContext.httpMethod',
+        'requestContext.identity.userAgent',
+        'requestContext.requestId',
+        'requestContext.resourcePath',
+        'requestContext.stage',
+        'resource',
+    ]
+    required_keys = [
+        'headers',
+        'httpMethod',
+        'path',
+        'requestContext',
+        'resource',
+    ]
+
+
+class CloudFront(EventType):
+    type = 'cloudFront'
+    keys = [
+        'Records[0].cf.config.distributionId',
+        'Records[0].cf.request.clientIp',
+        'Records[0].cf.request.headers.host[0].value',
+        'Records[0].cf.request.headers.user-agent[0].value',
+        'Records[0].cf.request.method',
+        'Records[0].cf.request.uri',
+    ]
+    required_keys = ['Records.cf']
```

Comment:
I just was poking around at this and saw that this maybe should be `required_keys = ['Records[0].cf']` ? Not sure just a heads up :)

Reply:
It will work either way, but in the interest of being explicit will update.